### ISSUE TITLE
Bug on FF51: Findbar doesn't get focus.

### ISSUE THREAD

**Issue by mzso** [Quicksaver/FindBar-Tweak#268]
Hi!

I open the quick findbar as usual (via keyconfig: gFindBar.startFind(gFindBar.FIND_TYPEAHEAD);), but when I type nothing happens because apparently the page still has the focus.

**Comment by Quicksaver** [comment]
That's strange, what Firefox version are you on?

**Comment by mzso** [comment]
[**@Quicksaver**](https://github.com/Quicksaver) commented on [2016. szept. 4. 23:48 CEST](https://github.com/Quicksaver/FindBar-Tweak/issues/268#issuecomment-244630710 "2016-09-04T21:48:45Z - Replied by Github Reply Comments"):
> That's strange, what Firefox version are you on?

As I said in the bug title it's FF51.
I also tried the built in hotkeys (alt+num /; shitf+6 on my keyboard). Same thing.

If I press a hotkey twice the findbar does get focuse but the text it had is erased.

**Comment by mzso** [comment]
Also it only happens when I have findbar tweaks enabled.

**Comment by Quicksaver** [comment]
Oh sorry, completely missed it! I'll take a look.

**Comment by mzso** [comment]
I forgot to mention that I'm on Windows (8.1). I'm not experiencing such bugs as the findbar not appearing that OSX users reported. (#263)

By the way did you look into it yet?

**Comment by mzso** [comment]
Hmmm... There's some breakage on Aurora too, before the new highlighting landed. At least the "highlight all matches by default" option doesn't work for me.

**Comment by mzso** [comment]
Hi!
The not getting focus thing successfully reached aurora. So I guess it's not tied with the new search highlights after all.

**Comment by jgjake2** [comment]
Here is a quick fix for anyone still annoyed with this bug. It's kinda like using duct tape to fix a hole in a boat, but it works for now :/ 

Edit "**fbt@quicksaver/resource/modules/mFinder.jsm**" and edit "**Modules.LOADMODULE**" to look like this:

```
Modules.LOADMODULE = function() {
	findbar.init('mFinder',
		function(bar) {
			// load our ._remoteFinder now to prevent the original from loading (it still exists and is fully functional, we just don't want it to be used)
			Messenger.loadInBrowser(bar.browser, 'mFinder');

			deinitializeNativeFinder(bar);

			if(bar.browser.isRemoteBrowser) {
				bar.browser._remoteFinder = new RemoteFinder(bar.browser);
			} else {
				bar.browser._finder = new RemoteFinder(bar.browser);
			}

			// if a search is run or re-run and there are no matches, make sure the findbar is visible so this is perfectly clear
			Piggyback.add('mFinder', bar, 'onFindResult', function(data) {
				if(data.result == this.nsITypeAheadFind.FIND_NOTFOUND && data.storeResult && this.hidden) {
					if(this.open(this.FIND_TYPEAHEAD)) {
						this._setFindCloseTimeout();
					}
					this._updateStatusUI(data.result, data.findBackwards);
				}
			}, Piggyback.MODE_AFTER);

			// this would usually be set when setting bar.browser, but there's no need to do all of that just for registering it with Finder
			bar.browser.finder.addResultListener(bar);
			
			var origFindbarOpen = bar.browser.finder.onFindbarOpen;
			bar.browser.finder.onFindbarOpen = function(arg){
				if(origFindbarOpen) {
						return origFindbarOpen(arg);
				}
			};
			
			var origFindbarClose = bar.browser.finder.onFindbarClose;
			bar.browser.finder.onFindbarClose = function(arg){
				if(origFindbarClose) {
						return origFindbarClose(arg);
				}
			};

			// get the initial isValid status for this bar's browser
			Messenger.messageBrowser(bar.browser, 'IsValid');
		},
```

I'm not very motivated right now, but you should if "origFindbarOpen" and "origFindbarClose" are actually functions first and apply all the given arguments instead of just the first one.

Anyway, good luck.

**Comment by mzso** [comment]
@jgjake2 
Thanks for the fix. It works, and that's what matters most.

**Comment by mzso** [comment]
@jgjake2 The only thing I found to be not working is the automatic highlight all option.

**Comment by jgjake2** [comment]
@mzso What version of FF are you using? Because it's working for me. Is there anything in your browser console when you hit Ctrl+F? (Browser console: Ctrl + Shift + J)

**Comment by jgjake2** [comment]
@mzso Ok I did find a bug when navigating a site. I guess FBT wasn't getting the memo that the page needed to be re-highlighted. So go to the following files:

* `fbt@quicksaver\resource\modules\content\highlightByDefault.jsm`
* `fbt@quicksaver\resource\modules\content\highlights.jsm`
* `fbt@quicksaver\resource\modules\content\mFinder.jsm`

Look for a line that looks like:
`onLocationChange: function(aWebProgress, aRequest, aLocation) {`
or
`onLocationChange: function(aWebProgress, aRequest, aLocation, aFlags) {`

There should be a line that blocks iFrames from triggering a refresh:
`if(aWebProgress.isTopLevel) {`

remove that line (and the related closing bracket `}`)

This fixed the issue for me. In fact, the feature has never worked better :P

**Comment by mzso** [comment]
[**@jgjake2**](https://github.com/jgjake2) commented on [2016. szept. 27. 03:10 CEST](https://github.com/Quicksaver/FindBar-Tweak/issues/268#issuecomment-249741060 "2016-09-27T01:10:53Z - Replied by Github Reply Comments"):
> [@mzso](https://github.com/mzso) What version of FF are you using? Because it's working for me. Is there anything in your browser console when you hit Ctrl+F? (Browser console: Ctrl + Shift + J)

Aurora, at present. Also I normally use the quick findbar. Nothing appears in the console.

I'll try your new fix.

**Comment by mzso** [comment]
[**@jgjake2**](https://github.com/jgjake2) commented on [2016. szept. 27. 04:02 CEST](https://github.com/Quicksaver/FindBar-Tweak/issues/268#issuecomment-249747475 "2016-09-27T02:02:48Z - Replied by Github Reply Comments"):
> [@mzso](https://github.com/mzso) Ok I did find a bug when navigating a site. I guess FBT wasn't getting the memo that the page needed to be re-highlighted. So go to the following files:
> This fixed the issue for me. In fact, the feature has never worked better :P

This doesn't work for me, if I did it right. Still no highlights by default, or at all with quick findbar.
Can you attach the modified xpi just in case I borked the editing?

**Comment by jgjake2** [comment]
> Aurora, at present
Hmm, I use nightly. I don't know if that would make a difference with this issue though.

> This doesn't work for me, if I did it right. Still no highlights by default, or at all with quick findbar.
Can you attach the modified xpi just in case I borked the editing?

Sure. Download it [here](fonts.wikijs.org/fbt_highlight_fix__jgjake2.zip). It just includes the files changed files. So just copy/paste them in.

**Comment by mzso** [comment]
@jgjake2 No joy. Oh, well, I can live with this until Quicksaver updates the addon.

**Comment by mzso** [comment]
Seems like this was fixed on master not long ago.  Tried it and it works. Including the highlighting by default.

**Comment by mzso** [comment]
I suck. I got mixed up...
@jgjake2 I just realized the difference might be because you are probably on mozilla central, and I was on aurora. I moved to central today, and the highlights started to function normally.

**Comment by mzso** [comment]
@Quicksaver Did you loose interest in findbar tweak?

**Comment by Quicksaver** [comment]
No, I just haven't had much time to work on my add-ons, so I have to prioritize. I should get to FBT still this month. ;)

**Comment by kilara1988** [comment]
@jgjake2  Can you please upload the modified XPI file again?The previous link is not working.
I tried editing it myself, but FF disabled the addon because it was not verified.

**Comment by mzso** [comment]
[**@kilara1988**](https://github.com/kilara1988) commented on [2016. nov. 19. 16:42 CET](https://github.com/Quicksaver/FindBar-Tweak/issues/268#issuecomment-261721043 "2016-11-19T15:42:15Z - Replied by Github Reply Comments"):
> [@jgjake2](https://github.com/jgjake2)  Can you please upload the modified XPI file again?The previous link is not working.  
> 
> I tried editing it myself, but FF disabled the addon because it was not verified.

Here's the patched xpi:

[findbar_tweak-2.1.10-fx-mod2.zip](https://github.com/Quicksaver/FindBar-Tweak/files/601785/findbar_tweak-2.1.10-fx-mod2.zip)

**Comment by kilara1988** [comment]
Thanks,but FF says it can't install it because the file is corrupt.Do i have to use Nightly/Developer Version where the addon signing check is disabled?I am using FF Beta now.

**Comment by mzso** [comment]
[**@kilara1988**](https://github.com/kilara1988) commented on [2016. nov. 20. 00:24 CET](https://github.com/Quicksaver/FindBar-Tweak/issues/268#issuecomment-261746749 "2016-11-19T23:24:53Z - Replied by Github Reply Comments"):
> Thanks,but FF says it can't install it because the file is corrupt.Do i have to use Nightly/Developer Version where the addon signing check is disabled?I am using FF Beta now.

Of course you need signing disabled, it's not signed.
It's not corrupt. Just unpack it...

**Comment by kilara1988** [comment]
Can it be disabled on FF Beta?I have "xpinstall.signatures.required" set to False,but still FF won't let me install it.And btw i unpacked the archive and tried dragging the XPI file to FF or manually installing it from the Addons Manager,but i get the file corrupted message.I know its because it is not signed and not because it is corrupted.That is why i asked if i need Nightly or i can still disable the check in Beta?

**Comment by mzso** [comment]
[**@kilara1988**](https://github.com/kilara1988) commented on [2016. nov. 20. 00:34 CET](https://github.com/Quicksaver/FindBar-Tweak/issues/268#issuecomment-261747214 "2016-11-19T23:34:51Z - Replied by Github Reply Comments"):
> Can it be disabled on FF Beta?I have "xpinstall.signatures.required" set to False,but still FF won't let me install it.And btw i unpacked the archive and tried dragging the XPI file to FF or manually installing it from the Addons Manager,but i get the file corrupted message.I know its because it is not signed and not because it is corrupted.That is why i asked if i need Nightly or i can still disable the check in Beta?

I don't remember it claiming to be corrupted. Anyway, you can disable it on every channel but beta. (Unless they stopped being pussies and started enforcing it on stable.)

**Comment by kilara1988** [comment]
I am sorry,but how can i install the version which fixes a bunch of bugs from 1 day ago?It doesn't show up on Firefox addons page yet and i can't figure out how to install it from here either.

**Comment by Quicksaver** [comment]
I will post a beta version for you to try out soon.

**Comment by Quicksaver** [comment]
All issues should be fixed in [beta version 2.1.11b1](https://github.com/Quicksaver/FindBar-Tweak/releases/tag/v2.1.11b1). Please give it a try and let me know if you still have any problems.

**Comment by Quicksaver** [comment]
Also, I'm sorry it took me so long to get to this.

**Comment by talha131** [comment]
I can confirm, this issue is fixed in `beta version 2.1.11b1`. Thank you @Quicksaver
